pkg/common/config.go

```diff
@@ -21,9 +21,55 @@ package common
 // in 2019 the community found this point is not very reasonable, but has not changed
 // see https://github.com/spf13/viper/pull/758
 
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
 type Config struct {
-	Auth AuthConfig `mapstructure:"auth"`
-	Saas SaasConfig `mapstructure:"saas"`
+	Auth    AuthConfig    `mapstructure:"auth"`
+	Saas    SaasConfig    `mapstructure:"saas"`
+	Cluster ClusterConfig `mapstructure:"cluster"`
+}
+
+// copy from "k8s.io/client-go/tools/clientcmd/api"
+// Cluster contains information about how to communicate with a kubernetes cluster
+type ClusterConfig struct {
```

Comment:
seems this field is useless ?

Reply:
It is the content of the cluster section of the configuration file.

Reply:
Not used now, but should be used later.

Reply:
> Not used now, but should be used later.

1. Read from `config.yaml` and merge with viper
2. use this `Config` for initialization
- kube client initalization 
- ...